File: javascript/src/models/backgroundTaskOut.ts
// this file is @generated
/* eslint @typescript-eslint/no-explicit-any: 0 */
import {
  BackgroundTaskStatus,
  BackgroundTaskStatusSerializer,
} from "./backgroundTaskStatus";
import { BackgroundTaskType, BackgroundTaskTypeSerializer } from "./backgroundTaskType";

export interface BackgroundTaskOut {
  data: any;
  /** The QueueBackgroundTask's ID. */
  id: string;
  status: BackgroundTaskStatus;
  task: BackgroundTaskType;
}

export const BackgroundTaskOutSerializer = {
  _fromJsonObject(object: any): BackgroundTaskOut {
    return {
      data: object["data"],
      id: object["id"],
      status: BackgroundTaskStatusSerializer._fromJsonObject(object["status"]),
      task: BackgroundTaskTypeSerializer._fromJsonObject(object["task"]),
    };
  },

  _toJsonObject(self: BackgroundTaskOut): any {
    return {
      data: self.data,
      id: self.id,
      status: BackgroundTaskStatusSerializer._toJsonObject(self.status),
      task: BackgroundTaskTypeSerializer._toJsonObject(self.task),
    };
  },
};

File: kotlin/lib/src/main/kotlin/models/BackgroundTaskOut.kt
// This file is @generated
package com.svix.kotlin.models

import com.svix.kotlin.StringAnyMapSerializer
import kotlinx.datetime.Instant
import kotlinx.serialization.Serializable

@Serializable
data class BackgroundTaskOut(
    @Serializable(with = StringAnyMapSerializer::class) val data: Map<String, Any>,
    /** The QueueBackgroundTask's ID. */
    val id: String,
    val status: BackgroundTaskStatus,
    val task: BackgroundTaskType,
    val updatedAt: Instant,
)
